jenkins/master.sls

```diff
@@ -0,0 +1,21 @@
+{% from "jenkins/map.jinja" import jenkins with context %}
+
+create_ssh_directory_on_master:
+  file.directory:
+    - name: {{ jenkins.home }}/.ssh
+    - user: {{ jenkins.user }}
+    - group: {{ jenkins.group }}
+    - mode: 700
+
+# This is just a static key, during production dynamic key generation is
+# recommended
```

Comment:
For a start, you could use ``cmd.run`` with ``onlyif`` or ``unless`` to generate keys